k-distribution/src/main/scripts/lib/kit/utils.py

```diff
@@ -0,0 +1,153 @@
+from collections import Counter
+from pathlib import Path
+from typing import Any, Dict, Mapping
+
+from pyk.kast import (
+    KApply,
+    KConstant,
+    KVariable,
+    isKApply,
+    isKRewrite,
+    isKSequence,
+    isKVariable,
+)
+from pyk.kastManip import (
+    buildAssoc,
+    countVarOccurances,
+    rewriteAnywhereWith,
+    traverseTopDown,
+)
+
+
+def ml_top():
+    return KConstant('#Top')
+
+
+def ml_and(terms):
+    return buildAssoc(ml_top(), '#And', terms)
+
+
+def extract_claims(module):
+    return [claim for claim in module['localSentences'] if claim['node'] == 'KClaim']
+
+
+def eliminate_generated_top(term):
+    rule = KApply('<generatedTop>', [KVariable('CONFIG'), KVariable('_')]), KVariable('CONFIG')
+    return rewriteAnywhereWith(rule, term)
+
+
+def extract_lhs(term):
+    return traverseTopDown(term, if_rewrite(get_lhs))
+
```

Comment:
We should make a PR which cleans up `kastManip.py` according to the new style guide, and organizes it a bit (with headers explaining the sections). Bits that shouldn't exist there should find new homes.

Then these (and most of the rest of this file) can be migrated there.